### PR TITLE
Feature/fair 178

### PR DESCRIPTION
**Pull request by ame-test**
**What changes does this PR bring?**

**Definition of Done**
The following need to be checked off as they are complete as part of the
PR process. If you are unable to mark off an item as complete, then state
in the comments why this is the case.
- [ ] Have you built the code locally and confirmed its working?
- [ ] Is the build confirmed to be passing on CI?
- [ ] Have you added enough relevant unit tests for your change?
- [ ] Have you added enough relevant E2E tests for your change?
- [ ] Have you updated any relevant documentation as part of this change?
- [ ] Has this change been tested against the Acceptance Criteria?
- [ ] Have you considered how this will be deployed in SIT/Staging/Production?

### CODE REVIEW

File: src/app/api/uploadgar/validations.js [hunk 1]
```diff
@@ -97,6 +97,7 @@ module.exports.validations = (voyageObj, crewArr, passengersArr) => {
     validationArr.push([
       new ValidationRule(validator.notEmpty, '', crew.documentNumber, `Enter a document number for ${name}`),
       new ValidationRule(validator.isValidStringLength, '', crew.documentNumber, `Travel document number for ${name} must be ${MAX_STRING_LENGTH} characters or less`),
+      new ValidationRule(validator.isAlphanumeric, '', crew.documentNumber, `Travel document number for ${name} must consistent only of Alphanumeric characters e.g. a-z, A-Z and 0-9. Special characters and spaces are not allowed e.g. $ % ! ' '`),
```
Comment (ho-henry-senior) [inline src/app/api/uploadgar/validations.js:100]: I think validation message could be shorter, maybe: 'Travel document number for ${name} must be alphanumeric only.'

File: src/app/people/validations.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ module.exports.validations = (req) => {
       [
         new ValidationRule(validator.notEmpty, 'travelDocumentNumber', req.body.travelDocumentNumber, 'Enter the travel document number'),
         new ValidationRule(validator.isValidStringLength, 'travelDocumentNumber', req.body.travelDocumentNumber, `Travel document number must be ${MAX_STRING_LENGTH} characters or less`),
+        new ValidationRule(validator.isAlphanumeric, 'travelDocumentNumber', req.body.travelDocumentNumber, `Travel document number must consistent only of Alphanumeric characters e.g. a-z, A-Z and 0-9. No special characters are allowed e.g. $ % !`),
```
Comment (ho-henry-senior) [inline src/app/people/validations.js:41]: same for this one: 'Travel document number must be alphanumeric only.'

File: src/app/people/validations.js [hunk 3]
```diff
@@ -81,6 +82,7 @@ module.exports.validations = (req) => {
     [
       new ValidationRule(validator.notEmpty, 'travelDocumentNumber', req.body.travelDocumentNumber, 'Enter the travel document number'),
       new ValidationRule(validator.isValidStringLength, 'travelDocumentNumber', req.body.travelDocumentNumber, `Travel document number must be ${MAX_STRING_LENGTH} characters or less`),
+      new ValidationRule(validator.isAlphanumeric, 'travelDocumentNumber', req.body.travelDocumentNumber, `Travel document number consistent only of Alphanumeric characters e.g. a-z, A-Z and 0-9. Special characters and spaces are not allowed e.g. $ % ! ' '`),
```
Comment (ho-henry-senior) [inline src/app/people/validations.js:85]: same for this one: 'Travel document number must be alphanumeric only.'

File: src/common/utils/validator.js [hunk 4]
```diff
@@ -664,6 +664,12 @@ function invalidLongDirection(value) {
 
 }
 
+function isAlphanumeric(input) {
+  // Use a regular expression to match only alphanumeric characters
+  var alphanumericRegex = /^[a-zA-Z0-9]+$/;
```
Comment (ho-henry-senior) [inline src/common/utils/validator.js:669]: this could be declared `const`

Comment (ame-test) [reply src/common/utils/validator.js]: Fixed in code.

File: src/app/people/validations.js (hunk 2 above)
Comment (ame-test) [reply src/app/people/validations.js]: Changed error message.

File: src/app/api/uploadgar/validations.js (hunk 1 above)
Comment (ame-test) [reply src/app/api/uploadgar/validations.js]: changed error message.

**Review by ho-henry-senior — APPROVED** [review]

**Pull request merged**